web_ui/server_api.go

```diff
@@ -266,39 +330,123 @@ func HandleUpdateDowntime(ctx *gin.Context) {
+	if err := database.UpdateDowntime(uuid, &updatedDowntime); err != nil {
 		ctx.JSON(http.StatusInternalServerError, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to update downtime: " + err.Error()})
 		return
 	}
-	ctx.JSON(http.StatusOK, existingDowntime)
+	ctx.JSON(http.StatusOK, updatedDowntime)
 }
 
 func HandleDeleteDowntime(ctx *gin.Context) {
 	uuid := ctx.Param("uuid")
+	existingDowntime, err := database.GetDowntimeByUUID(uuid)
+	if err != nil {
+		ctx.JSON(http.StatusNotFound, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Downtime record not found"})
+		return
+	}
+
+	if err := mirrorDowntimeToRegistry(ctx, *existingDowntime, http.MethodDelete, uuid); err != nil {
+		ctx.JSON(http.StatusBadGateway, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to mirror downtime to registry: " + err.Error()})
+		return
+	}
+
 	if err := database.DeleteDowntime(uuid); err != nil {
 		ctx.JSON(http.StatusInternalServerError, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to delete downtime: " + err.Error()})
 		return
 	}
 	ctx.JSON(http.StatusOK, server_structs.SimpleApiResp{Status: server_structs.RespOK, Msg: "Downtime deleted successfully"})
 }
+
+// mirrorDowntimeToRegistry forwards a downtime CRUD to the Registry, if the
+// current server has an Origin or Cache service. It uses the server's issuer key to mint a short-lived
+// service token and sets Authorization: Bearer <token>.
+func mirrorDowntimeToRegistry(ctx *gin.Context, dt server_structs.Downtime, method string, id string) error {
+	if id == "" {
+		return errors.New("downtime ID is required")
+	}
+
+	// Only mirror downtime to Registry when this server is an Origin or Cache
+	if !validateServerType([]server_structs.ServerType{server_structs.OriginType, server_structs.CacheType}) {
+		return nil
+	}
+
+	fed, err := config.GetFederation(ctx)
+	if err != nil || fed.RegistryEndpoint == "" {
+		return errors.Wrap(err, "failed to load federation configuration")
+	}
+
+	regURL, err := url.Parse(fed.RegistryEndpoint)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse registry endpoint")
+	}
+
+	// In "federation in a box" scenario, all services run in the same process with the same URL.
+	// Skip mirroring if the Registry endpoint is the same as the current server to avoid recursion.
+	currentServerURL := param.Server_ExternalWebUrl.GetString()
+	if currentServerURL != "" {
+		currentURL, err := url.Parse(currentServerURL)
+		if err == nil && currentURL.Host == regURL.Host {
+			log.Debugf("Skipping mirror to registry because Registry endpoint (%s) is the same as current server (%s)", regURL.Host, currentURL.Host)
+			return nil
+		}
+	}
+
+	regURL.Path = path.Join(regURL.Path, "api", "v1.0", "downtime", id)
+
+	tokCfg := token.NewWLCGToken()
+	tokCfg.Lifetime = 2 * time.Minute
+	tokCfg.Subject = dt.ServerID
+	tokCfg.AddAudienceAny()
```

Comment:
In this case, we explicitly know who is going to use this token -- the Registry. Can you set the audience to the registry's discovered hostname:port?